### PR TITLE
[Bugfix] Sort/With Parameter Issue Combobox

### PR DESCRIPTION
**Pull request by Civolilah**
@beganovich @turbo124 The PR includes fixes for adding/removing 'sort' and 'with' parameters. Let me know your thoughts.

### CODE REVIEW

File: src/components/forms/Combobox.tsx [hunk 1]
```diff
@@ -883,6 +883,26 @@ export function ComboboxAsync<T = any>({
 
         if (inputOptions.value && inputOptions.value.toString().length > 0) {
           currentUrl.searchParams.set('with', inputOptions.value.toString());
+
+          if (currentUrl.searchParams.get('sort')) {
+            currentUrl.searchParams.delete('sort');
+          }
```
Comment (Civolilah) [inline src/components/forms/Combobox.tsx:889]: In the moment when the 'with' parameter is set, we need to ensure the removal of the 'sort' parameter if it exists.

File: src/components/forms/Combobox.tsx [hunk 2]
```diff
@@ -883,6 +883,26 @@ export function ComboboxAsync<T = any>({
...
+      setUrl((c) => {
+        const currentUrl = new URL(c);
+
+        if (currentUrl.searchParams.get('with')) {
+          currentUrl.searchParams.delete('with');
+        }
+
+        if (sortBy) {
+          currentUrl.searchParams.set('sort', sortBy);
         }
```
Comment (Civolilah) [inline src/components/forms/Combobox.tsx:906]: When removing/clearing a value from the combobox, it is essential to ensure that the 'sort' parameter is included to obtain accurate results without the 'with' parameter (default list).

File: src/components/forms/Combobox.tsx [hunk 3]
```diff
@@ -883,12 +883,32 @@ export function ComboboxAsync<T = any>({
...
```
Comment (Civolilah) [inline src/components/forms/Combobox.tsx:906]: When removing/clearing a value from the combobox, it is essential to ensure that the 'sort' parameter is included to obtain accurate results without the 'with' parameter (default list).

**Comment by beganovich** [comment]
I think it looks. Is it possible to make test coverage for this?

**Comment by Civolilah** [comment]
> Looks good. Is it possible to make test coverage for this?

@beganovich Sure, it's a bit tricky to test queries, but I managed to find a way to test those queries. It's implemented to handle cases where we should have the sort parameter but not the sort and reversed. Let me know your thoughts.

**Review by beganovich — APPROVED** [review]

**Pull request merged**